### ISSUE TITLE
Overrides field

### ISSUE THREAD

**Issue by cdfa** [igorshubovych/markdownlint-cli#53]
I think it would be great to specify config overrides for certain files (like overrides in eslint and prettier). I don't think this should be implement in the `markdownlint` library, because it's easier to do this overriding manually when using the API, but it would be nice if the CLI would support it in some way.

**Comment by DavidAnson** [comment]
Relevant example: https://eslint.org/docs/user-guide/configuring#disabling-rules-only-for-a-group-of-files

**Comment by DavidAnson** [comment]
@cdfa, in case you are not aware, when using the VS Code extension, it will allow a config file in subdirectories as well as the root of the project. That can be used to customize settings for an entire subdirectory of files. Would that work for your purposes?

**Comment by cdfa** [comment]
Sadly not, I need a different configuration for .mdx files (specifically: disabling the forbidden html tag rule)

**Comment by DavidAnson** [comment]
Would it work to run the CLI in two separate passes? The first for *.md and the second for *.mdx (with slightly different settings)?

**Comment by cdfa** [comment]
Yes, I have a mdx only config and passing that using the `--config` option, but I can't make it "inherit" the default .md config.

**Comment by DavidAnson** [comment]
To inherit rules, have you looked at `extends`? https://github.com/DavidAnson/markdownlint/blob/master/README.md#optionsconfig

**Comment by cdfa** [comment]
Ah, that makes my case possible. I still think it would be convenient to not have to pass the specific config for every extension, but this works. Thank you!

**Comment by DavidAnson** [comment]
Cool. I’ll leave this open here for the future, but glad you’re unblocked now!
